src/plot_api/helpers.js

```diff
@@ -102,13 +102,12 @@ exports.cleanLayout = function(layout) {
         }
     }
 
-    if(layout.annotations !== undefined && !Array.isArray(layout.annotations)) {
-        Lib.warn('Annotations must be an array.');
-        delete layout.annotations;
-    }
     var annotationsLen = (layout.annotations || []).length;
```

Comment:
done in https://github.com/plotly/plotly.js/pull/1118/commits/9de2f772e537c7e4fe9cd87458369804ac88a25c